### PR TITLE
cssp: Fix warnings (#4503)

### PR DESCRIPTION
**Pull request by mfleisz**
Fix warnings introduced in #4499 as reported in #4503.

### CODE REVIEW

File: libfreerdp/core/nla.c [hunk 1]
```diff
@@ -1129,7 +1129,7 @@ SECURITY_STATUS nla_encrypt_public_key_hash(rdpNla* nla)
 		goto out;
 
 	/* include trailing \0 from hashMagic */
-	if (!winpr_Digest_Update(sha256, hashMagic, strlen(hashMagic) + 1))
+	if (!winpr_Digest_Update(sha256, (const BYTE*)hashMagic, strlen(hashMagic) + 1))
```
Comment (akallabeth) [inline libfreerdp/core/nla.c]: Why not just declare the buffer as `BYTE` and use `ARRAYSIZE` for the length?

Comment (akallabeth) [reply libfreerdp/core/nla.c]: and if we are at cleanups, have a look, there are still a lot of broken/wrong/invalid `(int)` casts around

**Comment by mfleisz** [comment]
@akallabeth Changed the handling of the hash magic and fixed some int/size_t casts

File: libfreerdp/core/nla.c [hunk 2]
```diff
@@ -115,8 +115,26 @@ static void nla_identity_free(SEC_WINNT_AUTH_IDENTITY* identity);
 #define ber_sizeof_sequence_octet_string(length) ber_sizeof_contextual_tag(ber_sizeof_octet_string(length)) + ber_sizeof_octet_string(length)
 #define ber_write_sequence_octet_string(stream, context, value, length) ber_write_contextual_tag(stream, context, ber_sizeof_octet_string(length), TRUE) + ber_write_octet_string(stream, value, length)
 
-static const CHAR ClientServerHashMagic[] = "CredSSP Client-To-Server Binding Hash\0";
-static const CHAR ServerClientHashMagic[] = "CredSSP Server-To-Client Binding Hash\0";
+/* CredSSP Client-To-Server Binding Hash\0 */
+static const BYTE ClientServerHashMagic[] =
```
Comment (akallabeth) [inline libfreerdp/core/nla.c]: Does VS complain when initializing `BYTE` (aka `unsigned char`) simply as string?

Comment (mfleisz) [reply libfreerdp/core/nla.c]: VS 2017 has no complaint about it - but then there are other compilers out there as we know. Also this way we ensure that there is definitely just one '\0' at the end of the string - compilers are supposed to add one at the end when defining strings this way ... supposed to.

Comment (akallabeth) [reply libfreerdp/core/nla.c]: @mfleisz  ok, keeping the magic :)

**Pull request merged**